x-pack/plugins/security/server/authentication/api_keys.ts

```diff
@@ -131,6 +155,39 @@ export class APIKeys {
     return result;
   }
 
+  /**
+   * Tries to grant an API key for the current user.
+   * @param request Request instance.
+   */
+  async grant(request: KibanaRequest): Promise<GrantAPIKeyResult | null> {
```

Comment:
We chatted a bit about the function naming. Did you decide to keep as is or go with something like `grantAsInternalUser`?